sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -175,13 +175,54 @@ case class DescribeDatabase(
   }
 }
 
+/**
+ * The DDL command that creates a function.
+ * alias: the class name that implements the created function.
+ * resources: Jars, files, or archives which need to be added to the environment when the function
+ *            is referenced for the first time by a session.
+ * isTemp: indicates if it is a temporary function.
+ */
+// TODO: Use Seq[FunctionResource] instead of Seq[(String, String)] for resources.
 case class CreateFunction(
```

Comment:
can we move the function related commands out, i.e.

functions.scala

this file is going to become thousands of lines long.